torch_geometric/loader/neighbor_loader.py

```diff
@@ -72,15 +77,23 @@ def __init__(
             self.colptr_dict, self.row_dict, self.perm_dict = out
 
             self.node_types, self.edge_types = data.metadata()
-            if isinstance(num_neighbors, (list, tuple)):
-                num_neighbors = {key: num_neighbors for key in self.edge_types}
-            assert isinstance(num_neighbors, dict)
-            self.num_neighbors = {
-                edge_type_to_str(key): value
-                for key, value in num_neighbors.items()
-            }
+            if not has_edges(data):
+                self.num_hops = 0
+                self.num_neighbors = {}
+            else:
+                if isinstance(num_neighbors, (list, tuple)):
+                    num_neighbors = {
+                        key: num_neighbors
+                        for key in self.edge_types
+                    }
+                assert isinstance(num_neighbors, dict)
+                self.num_neighbors = {
+                    edge_type_to_str(key): value
+                    for key, value in num_neighbors.items()
+                }
 
-            self.num_hops = max([len(v) for v in self.num_neighbors.values()])
+                self.num_hops = max(
```

Comment:
```suggestion
                # Add at least one element to the list to ensure that max is well-defined
                self.num_hops = max([0] +
```
to avoid the branching

Reply:
By avoiding branching do you mean remove the `if not has_edges(data)` condition?

Reply:
Yes

Reply:
I like to keep the branching statement to make it explicit that we do this when the graph has no edges.

Reply:
I prefer the comment too, makes it more clear why we even need to set it to be zero.